app.js

```diff
@@ -10,110 +17,241 @@ var product1Image = document.getElementById('img1');
 var product2Image = document.getElementById('img2');
 var product3Image = document.getElementById('img3');
 
-//I'm going to create a constructor function to change the images WHEN ONE of th ebuttons is clicked.
-//What do I want to change? The source.
 
-function Product(src, id) {
+function Product(src, name) {
   this.src = src;
-  this.id = id;
+  this.name = name;
+  this.timesDisplayed = 0;
   this.votes = 0;
 
 }
 
 var allProducts = [
-  new Product('img/bag.jpg'),
-  new Product('img/banana.jpg'),
-  new Product('img/bathroom.jpg'),
-  new Product('img/boots.jpg'),
-  new Product('img/breakfast.jpg'),
-  new Product('img/bubblegum.jpg'),
-  new Product('img/chair.jpg'),
-  new Product('img/cthulhu.jpg'),
-  new Product('img/dog-duck.jpg'),
-  new Product('img/dragon.jpg'),
-  new Product('img/pen.jpg'),
-  new Product('img/pet-sweep.jpg'),
-  new Product('img/scissors.jpg'),
-  new Product('img/shark.jpg'),
-  new Product('img/sweep.png'),
-  new Product('img/tauntaun.jpg'),
-  new Product('img/unicorn.jpg'),
-  new Product('img/usb.gif'),
-  new Product('img/water-can.jpg'),
-  new Product('img/wine-glass.jpg')
+  new Product('img/bag.jpg','R2D2 Bag'),
+  new Product('img/banana.jpg','Banana slicer'),
+  new Product('img/bathroom.jpg','Bathroom iPad'),
+  new Product('img/boots.jpg', 'Rain boots'),
+  new Product('img/breakfast.jpg', 'Breakfast machine'),
+  new Product('img/bubblegum.jpg', 'Bubble Gum balls'),
+  new Product('img/chair.jpg', 'Red chair'),
+  new Product('img/cthulhu.jpg', 'Green Monster'),
+  new Product('img/dog-duck.jpg', 'Dog-Duck'),
+  new Product('img/dragon.jpg', 'Dragon meat'),
+  new Product('img/pen.jpg', 'Pen lid'),
+  new Product('img/pet-sweep.jpg', 'Pet Sweep'),
+  new Product('img/scissors.jpg', 'Pizza scissors'),
+  new Product('img/shark.jpg', 'Shark sleeping bag'),
+  new Product('img/sweep.png', 'Baby sweep'),
+  new Product('img/tauntaun.jpg', 'Tauntaun'),
+  new Product('img/unicorn.jpg','Unicorn meat'),
+  new Product('img/usb.gif', 'Octo USB'),
+  new Product('img/water-can.jpg','Water can'),
+  new Product('img/wine-glass.jpg','Wine Glass')
 
 ];
-//Here I'm saying that these variables are these pictures.
+
 var product1 = allProducts[0];
 var product2 = allProducts[1];
 var product3 = allProducts[2];
 
-//I want to know how many votes does each button get WHEN CLICKED!!!
-//I want a function that tracks that.
-//Where am i listening: I'm listening at buttonNumber1
+Product.uniqueDouble = function () {
+  var uniqueOptions = [];
+  //math.random to generate random number
```

Comment:
These comments to understand what the code is doing are useful! I'd recommend going through other parts of your code in a similar way; this will help you gain a greater understanding of how this whole file fits together.